src/components/FilterView.js

```diff
@@ -0,0 +1,119 @@
+      <li className="list-group-item">
+        <div className="mt-1 mb-1 filter-name">Top Brands</div>
+        <div className="d-inline-flex flex-column">
+          {topBrands.map((v) => {
+            return (
+              <div key={v} className="form-check">
+                <input className="form-check-input" type="checkbox" onChange={(e) => onFilterChange(e,FILTER_BRAND,v)} value={filters[FILTER_BRAND].has(v)}/>
+                <label className="form-check-label" htmlFor="flexCheckDefault">
+                <span className="filter-element">{v}</span>
+                </label>
+              </div>
+            );
+          })}
+        </div>
+      </li>
+      <li className="list-group-item">
+      <div className="mt-1 mb-1 filter-name">Size</div>
+        <div className="d-flex flex-column">
+          {size.map((v) => {
+            return (
+              <div key={v} className="form-check">
+                <input className="form-check-input" type="checkbox" onChange={(e) => onFilterChange(e,FILTER_SIZE,v)} value={filters.size}/>
+                <label className="form-check-label" htmlFor="flexCheckDefault">
+                  <span className="filter-element">{v}</span>
+                </label>
+              </div>
+            );
+          })}
+        </div>
+      </li>
+      <li className="list-group-item">
+      <div className="mt-1 mb-1 filter-name">Ideal For</div>
+        <div className="d-flex flex-column">
+          {idealFor.map((v) => {
+            return (
+              <div key={v} className="form-check">
+                <input className="form-check-input" type="checkbox"  onChange={(e) => onFilterChange(e,FILTER_IDEAL_FOR,v)} value={"a"}/>
+                <label className="form-check-label" htmlFor="flexCheckDefault">
+                <span className="filter-element">{v}</span>
+                </label>
+              </div>
+            );
+          })}
+        </div>
+      </li>
+      <li className="list-group-item">
+      <div className="mt-1 mb-1 filter-name">Price Range</div>
+        <div className="d-grid d-block mb-3">
+          <div className="form-floating mb-2">
+            <input
+              type="number"
+              className="form-control border-radius-0"
+              value = {filters[FILTER_MIN_PRICE]}
+              onChange={(e) => onFilterChange(e, FILTER_MIN_PRICE,e.target.value)}
+            />
+            <label htmlFor="floatingInput">Min Price</label>
+          </div>
+          <div className="form-floating mb-2">
+            <input
+              type="number"
+              className="form-control border-radius-0"
+              value = {filters[FILTER_MAX_PRICE]}
+              onChange={(e) => onFilterChange(e, FILTER_MAX_PRICE,e.target.value)}
+            />
+            <label htmlFor="floatingInput">Max Price</label>
+          </div>
+          <button className="btn btn-dark border-radius-0" onClick={()=>{updateFilters(filters)}}>Apply</button>
```

Comment:
@rahulmeghwal I saw some repetition with the 3 button classes. Do you think it makes sense to abstract a higher-level class to encapsulate this avoid the duplication?

Reply:
I used bootstrap classes here so not sure if there is a way( using standard css) to derive new classes using existing css classes. Let me know if I misunderstood your comment.